Implement `odgi` package

tbhaxor opened the pull request:
This commit re-implements odgi package (superseded by #38741)

tbhaxor committed:
feature (packages): implement `odgi` package

This commit re-implements odgi package (superseded by #38741)

EbiArnie commented on var/spack/repos/builtin/packages/odgi/package.py:
```diff
@@ -0,0 +1,50 @@
+    )
+
+    # <<< Dependencies list starts here
+    requires("%gcc@9.3:12", when="%gcc")
```
This is covered by the lines above. It should be removed.

tbhaxor replied:
Done!

tldahlgren reviewed (commented):
Confirmed the home page and version commit.

tbhaxor committed:
fix (packages): remove redundant `requires()` from odgi package

This commit removes redundant use of `requires()` for gcc version in the `odgi` package

EbiArnie reviewed (approved):
OK. I built the binary with this and ran 'odgi test', seems to work.
